raven/senders/lua-resty-http.lua

```diff
@@ -0,0 +1,79 @@
+-- vim: st=4 sts=4 sw=4 et:
+--- Network backend using [lua-resty-http](https://github.com/ledgetech/lua-resty-http).
+--- Supports https, http, and keepalive for better performance.
+--
+-- @module raven.senders.lua-resty-http
+-- @copyright 2014-2017 CloudFlare, Inc.
+-- @license BSD 3-clause (see LICENSE file)
+
+local util = require 'raven.util'
+local http = require 'resty.http'
+local cjson = require 'cjson'
+
+local tostring = tostring
+local setmetatable = setmetatable
+local table_concat = table.concat
+local parse_dsn = util.parse_dsn
+local generate_auth_header = util.generate_auth_header
+local _VERSION = util._VERSION
+local _M = {}
+
+local mt = {}
+mt.__index = mt
+
+function mt:send(json_str)
+    local httpc = http.new()
+    local res, err = httpc:request_uri(self.server, {
+        method = "POST",
+        headers = {
+            ['Content-Type'] = 'applicaion/json',
+            ['User-Agent'] = "raven-lua-http/" .. _VERSION,
+            ['X-Sentry-Auth'] = generate_auth_header(self),
+            ["Content-Length"] = tostring(#json_str),
+        },
+        body = cjson.encode(json_str),
+        keepalive = self.opts.keepalive,
+        keepalive_timeout = self.opts.keepalive_timeout,
+        keepalive_pool = self.opts.keepalive_pool
+    })
+
+    if not res then
+        return nil, table_concat(err)
+    end
+
+    return true
```

Comment:
In this case the server responds a non-200 answer, the error would be silently ignored here. The `raven` library attempts to write the message in the error log as a last resort in case the sender cannot send it (https://github.com/cloudflare/raven-lua/blob/master/raven/init.lua#L323).